fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -555,20 +556,41 @@ protected void buildWhereClause(StringBuilder whereClause, String overrideType)
                     whereClause.append(whereClauseSegment);
                 } else {
                     if (!Type.COMPOSITE.equals(param.getType())) {
-                        // Join a standard parameter table
-                        //   JOIN Observation_TOKEN_VALUES AS param0
-                        //     ON param0.PARAMETER_NAME_ID=1191 AND param0.TOKEN_VALUE = :p1
-                        //    AND param0.LOGICAL_RESOURCE_ID = LR.LOGICAL_RESOURCE_ID
-
                         final String paramTableAlias = "param" + i;
-                        final String onFilter = querySegment.getQueryString().replaceAll(PARAMETER_TABLE_ALIAS + "\\.", paramTableAlias + ".");
-
-                        whereClause.append(JOIN);
-                        whereClause.append(tableName(overrideType, param));
-                        whereClause.append(" AS " + paramTableAlias);
-                        whereClause.append(ON);
-                        whereClause.append(onFilter);
-                        whereClause.append(" AND LR.LOGICAL_RESOURCE_ID = " + paramTableAlias + ".LOGICAL_RESOURCE_ID");
+                        if (param.isReverseChained()) {
+                            // Join on a select from logical resource table
```

Comment:
from Observation logical resources table. There's also a global (all resources) logical_resources table. Good to distinguish, because the join predicate would be slightly different

Reply:
comment updated